base/task.jl

```diff
@@ -208,6 +225,7 @@ function produce(v)
         else
             schedule_and_wait(t, v)
         end
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
True - the C code is actually easy to write, but it takes me a few weeks to nail this bug.